SimG4CMS/Forward/test/SimG4FluxAnalyzer.cc

```diff
@@ -0,0 +1,167 @@
+// system include files
+#include <memory>
+#include <iostream>
+#include <fstream>
+#include <vector>
+#include <map>
+#include <string>
+
+// user include files
+#include "CommonTools/UtilAlgos/interface/TFileService.h"
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+
+#include "SimDataFormats/CaloTest/interface/ParticleFlux.h"
+
+// Root objects
+#include "TROOT.h"
+#include "TSystem.h"
+#include "TFile.h"
+#include "TTree.h"
+
+//#define EDM_ML_DEBUG
+
+class SimG4FluxAnalyzer : public edm::one::EDAnalyzer<edm::one::WatchRuns,edm::one::SharedResources> {
+
+public:
+  explicit SimG4FluxAnalyzer(edm::ParameterSet const&);
+  ~SimG4FluxAnalyzer();
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  virtual void beginJob() override;
+  virtual void beginRun(edm::Run const&, edm::EventSetup const&) override {}
+  virtual void endRun(edm::Run const&, edm::EventSetup const&) override {}
+  virtual void analyze(edm::Event const&, edm::EventSetup const&) override;
+
+  edm::Service<TFileService>                   fs_;
+  std::vector<std::string>                     lvNames_;
+  std::vector<edm::EDGetTokenT<ParticleFlux> > tok_PF_;
+  TTree                                       *tree_;
+
+  std::vector<std::string>                     detName;
```

Comment:
@bsunanda - please, use consistent naming convention: trailing underscore for private variables